spring-web-modules/spring-mvc-basics-4/src/main/java/com/baeldung/optionalpathvars/ArticleViewerController.java

```diff
@@ -1,23 +0,0 @@
-package com.baeldung.optionalpathvars;
-
-import static com.baeldung.optionalpathvars.Article.DEFAULT_ARTICLE;
-
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
-
-@RestController
-public class ArticleViewerController {
-
-    @RequestMapping(value = {"/article", "/article/{id}"})
-    public Article getArticle(@PathVariable(name = "id") Integer articleId) {
```

Comment:
where have we moved this controller? was this not in use?

Reply:
This is basically covered in this Controller: `ArticleViewerWithRequiredAttributeController`. There are mapped two paths but in case you don't provide an `id`, you will receive an error. What do you think?

Reply:
required is set as false, even then it will throw an error?

Reply:
If I set the required to false for id it will not throw any error.